test/kubetest/k8s.go

```diff
@@ -582,48 +585,118 @@ func (l *K8s) DeletePodsForce(pods ...*v1.Pod) {
 	}
 }
 
-func (k8s *K8s) GetLogs(pod *v1.Pod, container string) (string, error) {
-	getLogsOpt := &v1.PodLogOptions{}
-	if len(container) > 0 {
-		getLogsOpt.Container = container
-	}
-	return k8s.GetLogsWithOptions(pod, getLogsOpt)
-}
-func (k8s *K8s) GetLogsWithOptions(pod *v1.Pod, options *v1.PodLogOptions) (string, error) {
-	var wg sync.WaitGroup
-	var result []byte
-	var err error
-	wg.Add(1)
+// GetLogsChannel returns logs channel from pod with the given options
+func (l *K8s) GetLogsChannel(ctx context.Context, pod *v1.Pod, options *v1.PodLogOptions) (chan string, chan error) {
+	options.Follow = true
+
+	linesChan := make(chan string, 1)
+	errChan := make(chan error, 1)
 	go func() {
-		defer wg.Done()
-		response := k8s.clientset.CoreV1().Pods(k8s.namespace).GetLogs(pod.Name, options)
-		result, err = response.DoRaw()
+		defer close(linesChan)
+		defer close(errChan)
+
+		reader, err := l.clientset.CoreV1().Pods(l.namespace).GetLogs(pod.Name, options).Stream()
+		if err != nil {
+			logrus.Errorf("Failed to get logs from %v", pod.Name)
+			errChan <- err
+			return
+		}
+		defer func() { _ = reader.Close() }()
+
+		scanner := bufio.NewScanner(reader)
+		for scanner.Scan() {
+			select {
+			case <-ctx.Done():
+				return
+			case linesChan <- scanner.Text():
```

Comment:
Does `scanner` returns text by line? Or just text available for reading?

Reply:
By line.
```
// Text returns the most recent token generated by a call to Scan
// as a newly allocated string holding its bytes.
```